### PR TITLE
feat: add field to define title for resultlist details

### LINKED ISSUE #957: Add a title for detail section of the resultlist
image

- [ ] Add a free input where you type your title
- [ ] Add the possibility to insert a field
- [ ] The proposed field should be keywords only
- [ ] when saving the dashbord, at export, add a new attribute to `fieldsConfiguration` as follow: 
image

- [ ] at import, read `fieldsConfiguration.detailsTitleTemplate`. If not defined, keep the detail control empty

NOTE :

- we can reuse the thumbnail template control in "Render" section of configuring a resultlist : 
image

### PR DESCRIPTION
**Pull request by sebbousquet**
- Fix #957

### CODE REVIEW

**Comment by sebbousquet** [comment]
New release of toolkit is needed before merge it

**Review by QuCMGisaia — APPROVED** [review]

File: src/app/modules/analytics-config/components/resultlist-data/resultlist-data.component.ts [hunk 1]
```diff
@@ -27,11 +32,48 @@ import { FormGroup } from '@angular/forms';
 export class ResultlistDataComponent implements OnInit {
 
   @Input() public control: FormGroup;
+  public detailsTitleControl: FieldTemplateControl;
```
Comment (MohamedHamouGisaia) [inline src/app/modules/analytics-config/components/resultlist-data/resultlist-data.component.ts:35]: Why not to declare it directly in `ResultlistConfigForm` ?

File: src/app/modules/analytics-config/components/resultlist-data/resultlist-data.component.ts [hunk 2]
```diff
@@ -27,11 +32,48 @@ import { FormGroup } from '@angular/forms';
...
+      {
+        optional: true
+      }
+    );
+
+    if (!!this.control.get('detailsTitle').value) {
+      this.detailsTitleControl.setValue(this.control.get('detailsTitle').value);
+    }
 
+    this.detailsTitleControl.valueChanges.subscribe({
```
[the remark this answers is not in the document]
Comment (sebbousquet) [reply src/app/modules/analytics-config/components/resultlist-data/resultlist-data.component.ts]: done

File: src/app/modules/analytics-config/components/resultlist-data/resultlist-data.component.ts [hunk 3]
```diff
@@ -27,11 +32,48 @@ import { FormGroup } from '@angular/forms';
...
+      next: (value) => {
+        this.control.get('detailsTitle').setValue(value);
+      }
+    });
+
+    if (!!collectionControl) {
+      collectionControl.valueChanges.subscribe(c => {
```
[the remark this answers is not in the document]
Comment (sebbousquet) [reply src/app/modules/analytics-config/components/resultlist-data/resultlist-data.component.ts]: done

**Review by MohamedHamouGisaia — APPROVED** [review]

**Pull request merged**